Terraform/Home_Task_2/modules/asg/main.tf

```diff
@@ -42,73 +43,52 @@ resource "aws_autoscaling_group" "this_asg" {
   max_size             = 4
   desired_capacity     = 2
   vpc_zone_identifier  = [var.private_subnet_1_id, var.private_subnet_2_id]
-  #availability_zones = ["us-east-1a"]
-  target_group_arns = [aws_alb_target_group.this_tg.arn]
+}
 
-  lifecycle {
-    ignore_changes = [load_balancers, target_group_arns] #[aws_lb.alb.arn, aws_alb_target_group.this_tg.arn]
-  }
-  depends_on = [aws_lb.alb]
+resource "aws_autoscaling_attachment" "asg_attachment_bar" {
+  autoscaling_group_name = aws_autoscaling_group.this_asg.id
+  lb_target_group_arn    = var.tg_arn
 }
 
-#-------------------------------------application load balancer---------------------------------------
-resource "aws_lb" "alb" {
-  name               = "lb-terraform"
-  internal           = false
-  load_balancer_type = "application"
-  ip_address_type    = "ipv4"
-  security_groups    = [var.alb_sg]
-  subnets            = [var.public_subnet_1_id, var.public_subnet_2_id]
+resource "aws_security_group" "sg_public" {
+  name   = "sg_pub_1"
+  vpc_id = var.vpc_id
 
-  enable_deletion_protection = false
-  /*
-  access_logs {
-    bucket  = aws_s3_bucket.lb_logs.bucket
-    prefix  = "test-lb"
-    enabled = true
+  ingress {
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]

```

Comment:
cidr_blocks, from_port, to_port - define values of these parameters in general variables.tf file